cpp/include/tensorrt_llm/batch_manager/cacheTransceiver.h

```diff
@@ -179,6 +211,32 @@ class CacheTransceiverComm
     c10::intrusive_ptr<c10d::ProcessGroup> mPgComm;
 };
 
+class UniqueIdClient
+{
+public:
+    static UniqueIdClient& instance()
+    {
+        static UniqueIdClient instance;
+        return instance;
+    }
+
+    int getUniqueId(std::string const& serverEndpoint, RequestIdType const& generationRequestId,
+        UuidType const& serverUuid, int32_t expectedRefCount);
```

Comment:
I think we should start using "sender/receiver" instead of "context/generation". Also, instead of "generationRequestId" should we use a more generic name such as "receiverTransferId"? For now, the transferID could be just the requestId, but eventually, if we have a separate API for KV cache transfer, there would not necessarily be a requestId...